### PR TITLE
more components tsx porting, fix a bug

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**

### CODE REVIEW

File: pkg/lib/cockpit-components-form-helper.tsx [hunk 1]
```diff
@@ -24,8 +24,8 @@ import { HelperText, HelperTextItem } from "@patternfly/react-core/dist/esm/comp
 
 export const FormHelper = ({ helperText, helperTextInvalid, variant, icon, fieldId } :
   {
-      helperText?: string,
-      helperTextInvalid?: string,
+      helperText?: string | null,
+      helperTextInvalid?: string | null,
```
Comment (martinpitt) [inline pkg/lib/cockpit-components-form-helper.tsx:28]: Ah, the `?` doesn't allow `null` by itself? That's rather strict. (I see it making sense both ways, TIL)

File: pkg/lib/cockpit-components-context-menu.tsx [hunk 2]
```diff
@@ -27,33 +26,36 @@ import "context-menu.scss";
...
-export const ContextMenu = ({ parentId, children }) => {
+export const ContextMenu = ({ parentId, children } : {
+    parentId: string,
+    children?: React.ReactNode[],
+}) => {
     const [visible, setVisible] = React.useState(false);
-    const [event, setEvent] = React.useState(null);
-    const root = React.useRef(null);
+    const [event, setEvent] = React.useState<MouseEvent | null>(null);
+    const root = React.useRef<HTMLDivElement>(null);
```
Comment (martinpitt) [inline pkg/lib/cockpit-components-context-menu.tsx:39]: Curious -- how can that get away without `HTMLDivElement | null`?

**Review by martinpitt — APPROVED** [review]
Nice! I like how the typing subsumes react proptypes. There's no runtime check, but there doesn't have to be.

File: pkg/lib/cockpit-components-context-menu.tsx [hunk 3]
```diff
@@ -27,33 +26,36 @@ import "context-menu.scss";
...
+            if (event && event.button === 0 && event.target instanceof HTMLElement) {
+                const wasOutside = !event.target.contains(root.current);
 
                 if (wasOutside)
                     setVisible(false);
             }
         };
 
-        const parent = document.getElementById(parentId);
+        const parent = document.getElementById(parentId)!;
```
Comment (jelly) [inline pkg/lib/cockpit-components-context-menu.tsx:58]: Why the bang at the end?

File: pkg/lib/cockpit-components-form-helper.tsx (hunk 1 above)
Comment (allisonkarlitskaya) [reply pkg/lib/cockpit-components-form-helper.tsx:28]: the `?` allows `undefined` but I've written code where it's more comfortable for this to be `null`.

File: pkg/lib/cockpit-components-context-menu.tsx (hunk 2 above)
Comment (allisonkarlitskaya) [reply pkg/lib/cockpit-components-context-menu.tsx:39]: There's some magic here where the resulting type is the union of the given type and the literal type of the thing passed in as the initial value.

**Pull request merged**